tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1

```diff
@@ -12,22 +12,47 @@ if ([string]::IsNullOrEmpty($outputFile)) {
 }
 
 $duplicates = @{}
-$environmentText = ''
+$envLines = @()
 
 try {
     foreach ($line in (Get-Content $outputFile)) {
         $duplicates[$line] = $true
-        $environmentText += ($line + "`n")
+        $envLines += $line
     }
-} catch {}
+}
+catch {}
 
 foreach ($entry in $DeploymentOutputs.GetEnumerator()) {
     $line = "$($entry.name)=$($entry.value)"
     if (!$duplicates.Contains($line)) {
         $duplicates[$line] = $true
-        $environmentText += ("$line" + "`n")
+        $envLines += $line
     }
 }
 
+$environmentText = $envLines -join "`n"
+
 $bytes = [System.Text.Encoding]::UTF8.GetBytes($environmentText)
 Set-Content $outputFile -Value $bytes -AsByteStream -Force
+
+function Write-Sourceable-BashFile([string]$sourceableBashFilePath) {
```

Comment:
Missing the envlines function parameter. Powershell scopes are annoying so it's pulling in the script scoped var anyways.

Reply:
Yeah, I wasn't sure if we "liked" that or not. I can also make the other one a function to reduce the scope.

Reply:
Oh, I forgot I left the parameter in. Yeah, I'll bring it back, I kind of liked the scoping.

Reply:
Bring it all back.